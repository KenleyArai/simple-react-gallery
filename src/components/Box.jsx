import styled from "styled-components";
import PropTypes from "prop-types";

import theme from "../theme";

const Box = styled.div`
  background: ${props => props.theme.primary};
`;

Box.propTypes = {
  theme: PropTypes.shape({
    primary: PropTypes.string,
    secondary: PropTypes.string,
    tertiary: PropTypes.string
  })
};

Box.defaultProps = {
  theme: { ...theme }
};

export default Box;
