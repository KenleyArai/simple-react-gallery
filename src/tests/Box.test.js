import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import "jest-styled-components";

configure({ adapter: new Adapter() });

import Box from "components/Box";

describe("Box", () => {
  it("should render box", () => {
    const wrapper = shallow(<Box />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with children", () => {
    const wrapper = shallow(<Box>Child Content</Box>);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with default primary background color", () => {
    const wrapper = shallow(<Box>Child Content</Box>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule("background", "#fff");
  });

  it("should render when provided a different primary background color", () => {
    const wrapper = shallow(
      <Box theme={{ primary: "#000" }}>Child Content</Box>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule("background", "#000");
  });
});
