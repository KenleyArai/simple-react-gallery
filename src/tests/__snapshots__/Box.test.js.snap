// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render box 1`] = `
<div
  className=""
/>
`;

exports[`Box should render with children 1`] = `
<div
  className=""
>
  Child Content
</div>
`;
